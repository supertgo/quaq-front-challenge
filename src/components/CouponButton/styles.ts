import styled, { css } from 'styled-components';

export const Wrapper = styled.button`
  ${({ theme }) => css`
    border: none;
    cursor: pointer;
    border-radius: ${theme.border.small};
    background-color: ${theme.colors.lightGreen};
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.xxsmall} ${theme.spacings.small};
    color: ${theme.colors.green};

    svg {
      width: 1.5rem;
      height: 1.5rem;
    }
  `}
`;
