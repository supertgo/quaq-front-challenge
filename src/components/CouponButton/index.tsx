import { ArrowIosForwardOutline as ArrowIcon } from '@styled-icons/evaicons-outline/ArrowIosForwardOutline';
import { Ticket } from '@styled-icons/ionicons-sharp/Ticket';

import * as S from './styles';

const CouponButton = () => (
  <S.Wrapper>
    <S.Content>
      <Ticket />
      <span>Clube Ifood pra quem ama cupom</span>
      <ArrowIcon />
    </S.Content>
  </S.Wrapper>
);

export default CouponButton;
