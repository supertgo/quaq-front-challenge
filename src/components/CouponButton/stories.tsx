import { Story, Meta } from '@storybook/react';
import CouponButton from '.';

export default {
  title: 'CouponButton',
  component: CouponButton
} as Meta;

export const Default: Story = () => <CouponButton />;
