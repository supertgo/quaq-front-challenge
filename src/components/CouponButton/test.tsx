import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helper';

import CouponButton from '.';

describe('<CouponButton />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<CouponButton />);

    expect(
      screen.getByText('Clube Ifood pra quem ama cupom')
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
