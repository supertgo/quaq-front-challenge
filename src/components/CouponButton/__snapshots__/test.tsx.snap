// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CouponButton /> should render the heading 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  border: none;
  cursor: pointer;
  border-radius: 0.4rem;
  background-color: #bff2de;
}

.c1 {
  padding: 0.8rem 2.4rem;
  color: #25790d;
}

.c1 svg {
  width: 1.5rem;
  height: 1.5rem;
}

<button
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M426.24 127.72l-10.94 10.94a29.67 29.67 0 01-42-42l10.94-10.94L314.52 16l-88 88-4 12.09-12.09 4L16 314.52l69.76 69.76 10.94-10.94a29.67 29.67 0 0142 42l-10.94 10.94L197.48 496l194.4-194.4 4-12.09 12.09-4 88-88zm-208.56 5.43l21.87-21.87 33 33-21.88 21.87zm43 43l21.88-21.88 32.52 32.52-21.88 21.88zm42.56 42.56l21.88-21.88 32.52 32.52-21.84 21.93zm75.57 75.56l-33-33 21.87-21.88 33 33z"
      />
    </svg>
    <span>
      Clube Ifood pra quem ama cupom
    </span>
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        data-name="Layer 2"
      >
        <path
          d="M10 19a1 1 0 01-.64-.23 1 1 0 01-.13-1.41L13.71 12 9.39 6.63a1 1 0 01.15-1.41 1 1 0 011.46.15l4.83 6a1 1 0 010 1.27l-5 6A1 1 0 0110 19z"
          data-name="arrow-ios-forward"
        />
      </g>
    </svg>
  </div>
</button>
`;
